### PR TITLE
Skaffold init support for polyglot-maven projects

### PR DESCRIPTION
**Pull request by MarlonGamez**
Fixes https://github.com/GoogleContainerTools/skaffold/issues/4335

Adds support for `skaffold init` to detect jib as a builder in different pom file extensions that are supported by https://github.com/takari/polyglot-maven.

I've added tests for a couple of extensions, but wasn't sure if I should for each case because I didn't want to clutter the tests with what is essentially just testing string matching

### CODE REVIEW

File: pkg/skaffold/build/jib/init.go [hunk 1]
```diff
@@ -89,13 +89,30 @@ func validate(path string, enableGradleAnalysis bool) []ArtifactConfig {
...
-	case strings.HasSuffix(path, "pom.xml"):
+	case isPomFile(path):
 		builderType = JibMaven
 		executable = "mvn"
 		wrapper = "mvnw"
 		searchString = "<artifactId>jib-maven-plugin</artifactId>"
 		taskName = "jib:_skaffold-init"
 		consoleFlag = "--batch-mode"
+
+		// Polyglot maven support: support pom files of different extension types
```
Comment (briandealwis) [inline pkg/skaffold/build/jib/init.go:100]: There's an easier and more robust way: polyglot maven [requires a `.mvn/extensions.xml` file](https://github.com/takari/polyglot-maven#usage).  Polyglot supports a bunch of other languages, including yaml 🙀

File: pkg/skaffold/build/jib/init.go [hunk 2]
```diff
@@ -89,13 +89,30 @@ func validate(path string, enableGradleAnalysis bool) []ArtifactConfig {
 	var builderType PluginType
 	var executable, wrapper, taskName, searchString, consoleFlag string
 	switch {
-	case strings.HasSuffix(path, "pom.xml"):
+	case isPomFile(path):
 		builderType = JibMaven
 		executable = "mvn"
 		wrapper = "mvnw"
 		searchString = "<artifactId>jib-maven-plugin</artifactId>"
```
Comment (briandealwis) [inline pkg/skaffold/build/jib/init.go:96]: TBH I think just searching for `jib-maven-plugin` is enough.  After all, this string would be fooled if someone had `jib-maven-plugin` as a `<dependency>`.

**Comment by MarlonGamez** [comment]
@briandealwis So do you think anything needs to be done beyond checking if something's a pom.* file and then searching for "jib-maven-plugin"?

**Review by briandealwis — APPROVED** [review]
I was going to say that we should check for the existence of `.mvn/extensions.xml`, but it could exist further up in the tree.  The chance of a pom.* that just happens to have the string `jib-maven-plugin` without being a Jib project is pretty slim.

**Pull request merged**